exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/Adapter.java

```diff
@@ -75,6 +75,10 @@ private Adapter() {}
     target.addAllLogs(toJaegerLogs(span.getEvents()));
     target.addAllReferences(toSpanRefs(span.getLinks()));
 
+    if (span.getResource() != null && span.getResource().getAttributes() != null) {
+      target.addAllTags(toKeyValues(span.getResource().getAttributes()));
```

Comment:
Resource corresponds to process, so we want to add these to the process's tags. Can you move this logic to the process here?

https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java#L89

Reply:
@anuraaga there I don't have `SpanData` to get `getResource().getAttributes()`.
should I get resource attributes using `Resource.getDefault().getAttributes()` ?

Reply:
Oh interesting - in that case I think this does need to be a bigger refactoring to move the process struct construction into the export flow. We should be able to follow a logic flow very similar to OTLP which starts by grouping by resource

https://www.github.com/open-telemetry/opentelemetry-java/tree/master/exporters%2Fotlp%2Fsrc%2Fmain%2Fjava%2Fio%2Fopentelemetry%2Fexporters%2Fotlp%2FSpanAdapter.java

Reply:
`Collector.PostSpansRequest` accepts only one `Model.Batch`. But `Model.Batch` is per `Model.Process`. 
https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java#L107
If there are several different Resources then I need to create several batches and send several requests.

Should I change [collector.proto](https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/jaeger/src/main/proto/jaeger/api_v2/collector.proto#L14) to accept multiple batches?
My concern is that jaeger backend will not accept such payload.

Reply:
Not an expert in this, but my guess is that we'll need to change the exporter to potentially send more than one batch, one per Resource on the span data. The case where you get multiple Resources per batch of spans should be very rare, so I wouldn't worry about the fact that we're sending multiple requests per export too much.

Reply:
refactored to group spans by resource and then set process tags per resource